src/testdir/test_channel.vim

```diff
@@ -1541,5 +1541,21 @@ func Test_close_lambda()
   call s:run_server('Ch_test_close_lambda')
 endfunc
 
+func Test_envs()
+  if !has('job')
+    return
+  endif
+
+  let s:envstr = ''
+  if has('win32')
+    call job_start(['cmd', '/c', 'echo %FOO%'], {'callback': {ch,msg->execute(":let s:envstr .= msg")}, 'envs':{'FOO': 'bar'}})
+  else
+    call job_start(['cmd', '/c', 'echo $FOO'], {'callback': {ch,msg->execute(":let s:envstr .= msg")}, 'envs':{'FOO': 'bar'}})
```

Comment:
I guess you wanted to use `[&shell, &shellcmdflag, 'echo $FOO']` instead of `cmd /c` that won't work on UNIX systems.